Dockerfile

```diff
@@ -0,0 +1,132 @@
+
+# Pip2/3 -- once we upgrade it's renamed pip3.6 --> pip3
+RUN pip install --upgrade pip
+RUN pip3.6 install --upgrade pip
+RUN pip3 install astropy
+RUN pip3 install ipython
+RUN pip3 install pyyaml
+RUN pip3 install fitsio==$FITSIO_VERSION
+
+# Make python3 link
+RUN ln -s /usr/bin/python3.6 /usr/bin/python3
+
+# ---------------------
+# ts_sal rpms
+# Add the lsst-ts repo
+RUN wget https://lsst-web.ncsa.illinois.edu/~felipe/packages/lsst-ts.repo
+RUN mv -v lsst-ts.repo /etc/yum.repos.d
+
+RUN yum -y install OpenSpliceDDS-$OSPL_VERSION
+RUN yum -y install ATHeaderService-$SAL_VERSION \
+    ATCamera-$SAL_VERSION \
+    ATArchiver-$SAL_VERSION \
+    EFD-$SAL_VERSION \
+    Scheduler-$SAL_VERSION \
+    ATPtg-$SAL_VERSION \
+    ATMCS-$SAL_VERSION \
+    ATSpectrograph-$SAL_VERSION \
+    ATTCS-$SAL_VERSION \
+    # Needed by ATArchiver
+    CatchupArchiver-$SAL_VERSION \
+    MTArchiver-$SAL_VERSION \
+    PromptProcessing-$SAL_VERSION
+
+
+# Get the setup conf
+RUN wget https://lsst-web.ncsa.illinois.edu/~felipe/packages/setup_SAL.env -O /opt/lsst/setup_SAL.env
+# ---------------------
+
+# Add $HSUSER as user
+RUN useradd -ms /bin/bash $HSUSER
+RUN usermod -aG wheel $HSUSER
+
+# Install HeaderService
+ARG PRODUCT=HeaderService
+ARG VERSION=$HEADERSERVICE_VERSION
+ARG PRODUCT_DIR=$INSTALL_PATH/$PRODUCT/$VERSION
+ARG INSTALL_DIR=$INSTALL_PATH/$PRODUCT/$VERSION/sandbox-install
+
+# Create the init env file
+RUN echo "source $INSTALL_PATH/setup_SAL.env" > $INSTALL_PATH/setup_HeaderService.env 
+
+# Git clone and version checkout
+RUN mkdir -p $INSTALL_DIR \
+    && cd $INSTALL_DIR \
+    && git clone $GIT_LSST/$PRODUCT.git \
+    && cd $PRODUCT \
+    && git checkout $VERSION \
+    && mkdir -p $PRODUCT_DIR \
+    && export PYTHONPATH=$PYTHONPATH:$PRODUCT_DIR/python \
+    && python3 setup.py install --prefix=$PRODUCT_DIR --install-lib=$PRODUCT_DIR/python
+
+# Add to the init file
+RUN echo "source $PRODUCT_DIR/setpath.sh $PRODUCT_DIR" >> $INSTALL_PATH/setup_HeaderService.env
+
+# Install the salpytools
+ARG PRODUCT=salpytools
+ARG VERSION=$SALPYTOOLS_VERSION
+ARG PRODUCT_DIR=$INSTALL_PATH/$PRODUCT/$VERSION
+ARG INSTALL_DIR=$INSTALL_PATH/$PRODUCT/$VERSION/sandbox-install
+
+# Git clone and version checkout
+RUN mkdir -p $INSTALL_DIR \
+    && cd $INSTALL_DIR \
+    && git clone $GIT_LSST/$PRODUCT.git \
+    && cd $PRODUCT \
+    && git checkout $VERSION \
+    && mkdir -p $PRODUCT_DIR \
+    && export PYTHONPATH=$PYTHONPATH:$PRODUCT_DIR/python \
+    && python3 setup.py install --prefix=$PRODUCT_DIR --install-lib=$PRODUCT_DIR/python
+
+# Add to the init file
+RUN echo "source $PRODUCT_DIR/setpath.sh $PRODUCT_DIR" >> $INSTALL_PATH/setup_HeaderService.env
+RUN echo "export LSST_DDS_DOMAIN=auxtelpath" >> $INSTALL_PATH/setup_HeaderService.env
+
+# Make /opt writable by $HSUSER
+RUN chown $HSUSER:$HSUSER /opt
+
+ENV USER $HSUSER
+ENV HOME /home/$HSUSER
+ENV SHELL /bin/bash
+
+USER $HSUSER
+WORKDIR /home/$HSUSER
+
+
```

Comment:
Is it intended that the HeaderService be running when this container is invoked?  If so, you'll need to add that.  I built the container and ran it, and didn't see the HeaderService running.   If not, make a note of how to start the HeaderService after the container is started, so people know how to do that.

Reply:
No, that's not the intention yet, so I'm adding a note to the docker file.